frontend/src/stores/roles/actions.ts

```diff
@@ -23,3 +23,23 @@ function setRoles(newRoles: Role[]): void {
 }
 
 export { fetchRoles, addRole }
+
+export const saveKnowledge = async (newKnowledge: string) => {
```

Comment:
You'll have to create an entire structure to knowledge state. Knowledge is a different entity than role. That's why, instead of put knowledge things inside `/src/stores/roles` folder, you'll have to create a new one (`/src/stores/knowledges`), and do the same thing

Reply:
done

Reply:
Done